ios/NotifeeCore/NotifeeCore+NSNotificationCenter.m

```diff
@@ -58,13 +58,18 @@ + (void)load {
 #pragma mark Application Notifications
 
 - (void)application_onDidFinishLaunchingNotification:(nonnull NSNotification *)notification {
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wdeprecated-declarations"
-  UILocalNotification *launchNotification =
-      (UILocalNotification *)notification.userInfo[UIApplicationLaunchOptionsLocalNotificationKey];
-  [[NotifeeCoreUNUserNotificationCenter instance]
-      onDidFinishLaunchingNotification:launchNotification.userInfo];
-  [[NotifeeCoreUNUserNotificationCenter instance] getInitialNotification];
+  NSDictionary *notifUserInfo =
+      notification.userInfo[UIApplicationLaunchOptionsLocalNotificationKey];
+    
+  if (!notifUserInfo) {
+      // Fallback to remote notification key if local notification key is not available
+      notifUserInfo = notification.userInfo[UIApplicationLaunchOptionsRemoteNotificationKey];
+  }
+
+  if (notifUserInfo) {
+      [[NotifeeCoreUNUserNotificationCenter instance] onDidFinishLaunchingNotification:notifUserInfo];
+      [[NotifeeCoreUNUserNotificationCenter instance] getInitialNotification];
+  }
```

Comment:
I ported just those changes in my project, and `await notifee.getInitialNotification();` hangs!
<img width="1261" alt="image" src="https://github.com/user-attachments/assets/55f44246-124f-4c7d-84b5-3c82818a4871">

```
    console.log('notifee.getInitialNotification-before');
    const initialNotification = await notifee.getInitialNotification();
    console.log('notifee.getInitialNotification-after'); <== not reached
```

Reply:
@dprevost-LMI  Hmm are you also using remote notifications? just double checking. Some debugger info from breakpoints would be great (that's how I solved my issue), but xcode debugging can be very hard if you never did it before. Learned it the hard way.

Reply:
> are you also using remote notifications?

We were for sure since it used to work, but I just took over a colleague's branch work for the whole Android 14 deadline; he might have "remove it"! I'll dig!

Reply:
At first, I also started with that change, and managed to make it partially work. But eventually I concluded that changing `application_onDidFinishLaunchingNotification()` would be the most elegant solution that fixed lots of issues for me. Please keep me posted!

Reply:
@dprevost-LMI By the way, be sure to set `notifee_options` like this: 

```
const message: MulticastMessage = {
  // ✅ We can continue using local notifcation for Android
  // 👍 while triggering iOS remote notifications from `apns`
  data: {notifee_options: JSON.stringify(notifeeOptions)}, <---- Maybe this oone too, but `apns` is more important
  tokens: [],
  android: {
    priority: 'high',
  },
  apns: {
    payload: {
      notifee_options: notifeeOptions, // <---- Notifee needs this here
      aps: {
        alert: {
          // 🚧 This is needed to trigger an alert/remote notification only for iOS
          // 👍 but Android will continue using data-only notifications
          title: 'Hello World',
        },
        mutableContent: true,
      },
    },
  },
};
```

You can see this is needed here:

https://github.com/invertase/notifee/blob/main/ios/NotifeeCore/NotifeeCoreUtil.m#L625

and many other places like: 

https://github.com/invertase/notifee/blob/main/ios/NotifeeCore/NotifeeCoreExtensionHelper.m#L43-L49

Reply:
I'll resolve this per request, but that commit you linked (that you are - I assume - locally reverting via patch-package was a bit touchy and there were issues for at least one other person. So that area of code + that change in particular is still on my "problematic behavior" list. I'm not proprietary about that list of course, anyone that wants to think deeply and propose a change is welcome and may beat me to it 😅

Reply:
I might try one day; I lack so much knowledge about that right now.
And yes, it is a patch locally! Mainly to meet Android 14 deadline dates!
But for sure, we will review it!

Reply:
---
> I tried to debug with Xcode on a working version, and even when using `Wait for the executable to be launched, the keys in question are nil.

1. That's why if **UIApplicationLaunchOptions<ins>Local</ins>NotificationKey** doesn't do the job, we try to use **UIApplicationLaunchOptions<ins>Remote</ins>NotificationKey** instead as a fallback. But if neither is working for you, I would expect you to hit # 3 in this list.

<br> 

----
> I cannot explain, but this is definitely related to the "remote" setup, where we have a second target named notifications that handles the reception and decryption of the notifications.

2. We're also using remote notifications by adding NotificationServiceExtension as a second target [as decribed in the docs](https://notifee.app/react-native/docs/ios/remote-notification-support). 

<br> 

---
> after reverting changes of commit [c9593cd](https://github.com/invertase/notifee/commit/c9593cded6dd98595a62d1eb9c2d23deb8cb445b) and testing on a release build, my background case and my close case started working.

3. In the beginning, I was also tempted to change this logic here, it really seemed to be the right thing to do, however, after finding # 1 in this list, I noticed I have the correct data flowing in `notifeeNotification` so no need to revert those for me anymore

Reply:
Last update, I was tired and did not see this, but for the close case, finally, on top of the patch, we had to keep the event in memory after the `onForegroundEvent` or `onBackgroundEvent` (both call the same function). Then on boot, after the redux event rehydrated, we process the tapped notification!